wheel/wheelfile.py

```diff
@@ -136,12 +142,19 @@ def write(self, filename, arcname=None, compress_type=None):
         zinfo.compress_type = ZIP_DEFLATED
         self.writestr(zinfo, data, compress_type)
 
+    def mkdir(self, filename, arcname):
+        st = os.stat(filename)
+        zinfo = ZipInfo(arcname + '/', date_time=get_zipinfo_datetime(st.st_mtime))
+        zinfo.external_attr = st.st_mode << 16
+        zinfo.compress_type = ZIP_DEFLATED
+        self.writestr(zinfo, b'')
+
     def writestr(self, zinfo_or_arcname, bytes, compress_type=None):
         ZipFile.writestr(self, zinfo_or_arcname, bytes, compress_type)
         fname = (zinfo_or_arcname.filename if isinstance(zinfo_or_arcname, ZipInfo)
                  else zinfo_or_arcname)
         logger.info("adding '%s'", fname)
-        if fname != self.record_path:
+        if fname != self.record_path and not fname.endswith('/'):
```

Comment:
Is this check really necessary?

Reply:
If you take it out, the tests fail because there are RECORD entries for the directories (with a SHA hash for an empty file).

Reply:
Okay, I see it's also used by the `mkdir()` method.